README.md

```diff
@@ -67,3 +67,78 @@ jobs:
           chmod +x dist/bin/*
           dist/bin/${{ matrix.bin }} --version
 ```
+
+### [`cabal-ghc-compat`](./actions/cabal-ghc-compat/action.yml)
+
+Check that the Cabal configuration file is compatible with the version of the
+Cabal library bundled with GHC.
+
+Many Haskell projects support compilation with older GHC versions to help
+users build the project even if they lack access to the latest GHC and Cabal
+packages, such as when using versions provided by their OS package manager
+on older systems. However, compatibility with older Cabal versions is often
+overlooked. By default, commands like cabal build use the Cabal library version
+bundled with the `cabal-install` tool to process configuration files (`.cabal`,
+`cabal.project{,.freeze}`). If a project's CI configuration only tests recent
+cabal-install versions, it may build with older GHC versions but only with
+newer Cabal versions. This undermines compatibility, as users with older Cabal
+versions will still be unable to build the project.
+
+This action verifies that the project is buildable with the Cabal library
+version [bundled with GHC][bundled]. These GHC/Cabal versions are typically
+available in the same OS package sets. Using this action to check GHC/Cabal
+compatibility helps prevent accidental failures to support older combinations
+that users may rely on.
+
+[bundled]: https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history
+
+This action performs this check by using Nix to install the specified version
+of GHC, creating a minimal `Setup.hs` file (which forces the project to be built
+using GHC's version of Cabal), and running `cabal clean` (which is sufficient
+to cause validation of the Cabal configuration files). Thus, it should be run
+either entirely before any `cabal {build,install,test}` steps or entirely after
+them. This strategy is more foolproof than explicitly including older Cabal
+versions in, e.g., a `matrix`, as it specifically checks the version of Cabal
+bundled with the specified version GHC.
+
+Inputs:
+
+- `dirs`: Directories in which to run
+- `ghc`: GHC version, e.g., "9.10.1"
+- `pkgs`: Extra packages to make available in the Nix shell (see action for
+  details and an example)
+- `token`: Set to `secrets.GITHUB_TOKEN`
+
+This action has no explicit outputs, but sets the `DO_IN_NIX_SHELL` environment
+variable to `nix shell ${GHC_NIXPKGS}#cabal-install ${GHC_NIXPKGS}#${GHC}
+${pkgs}` where `${GHC_NIXPKGS}` is a version of nixpkgs that contains the
+specified version of GHC from the `ghc` input, and ${pkgs}` is the list of
+packages from the `pkgs` input.
```

Comment:
Suggest describing why this is exported: "This exported variable allows for CI jobs that use this action to subsequently run the same cabal and GHC prepared by this action. For example, a job could then perform a build step via `${DO_IN_NIX_SHELL} cabal build`

Reply:
Good call, and thanks for the suggested wording! Added.